File: app/src/main/java/com/myhexaville/pagetransformer/PageTransformer.java
package com.myhexaville.pagetransformer;

import android.support.v4.view.ViewPager;
import android.view.View;


public class PageTransformer implements ViewPager.PageTransformer {
    private View firstImage, secondImage, thirdImage;

    @Override
    public void transformPage(View page, float position) {
        int pageWidth = page.getWidth();

        if ((int) page.getTag() == 0) {
            firstImage = page.findViewById(R.id.image_one);
            firstImage.setTranslationX(-position * (pageWidth / 2));
        }
        if ((int) page.getTag() == 1) {
            secondImage = page.findViewById(R.id.image_two);
            secondImage.setTranslationX(-position * (pageWidth / 2));
        }
        if ((int) page.getTag() == 2) {
            thirdImage = page.findViewById(R.id.image_three);
            thirdImage.setTranslationX(-position * (pageWidth / 2));
        }
    }
}

File: app/src/main/java/com/myhexaville/pagetransformer/PageTransformer.kt
package com.myhexaville.pagetransformer

import android.support.v4.view.ViewPager
import android.view.View


class PageTransformer : ViewPager.PageTransformer {
    var firstImage: View? = null
    var secondImage: View? = null
    var thirdImage: View? = null

    override fun transformPage(page: View?, position: Float) {
        val pageWidth = page?.width ?: return

        if (page.tag == 0) {
            firstImage = page.findViewById(R.id.image_one)
            firstImage?.translationX = -position * (pageWidth / 2)
        }
        if (page.tag == 1) {
            secondImage = page.findViewById(R.id.image_two)
            secondImage?.translationX = -position * (pageWidth / 2)
        }
        if (page.tag == 2) {
            thirdImage = page.findViewById(R.id.image_three)
            thirdImage?.translationX = -position * (pageWidth / 2)
        }
    }
}